packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -0,0 +1,295 @@
+    assert(padding != null);
+    assert(verticalOffset != null);
+    assert(screenEdgeMargin != null);
+    assert(preferBelow != null);
+    assert(fadeDuration != null);
+    assert(showDuration != null);
+  }
+
+  final String message;
+  final Color backgroundColor;
+  final Color textColor;
+  final TextStyle style;
+  final double opacity;
+  final double borderRadius;
+  final double height;
+  final EdgeDims padding;
+  final double verticalOffset;
+  final EdgeDims screenEdgeMargin;
+  final bool preferBelow;
+  final Duration fadeDuration;
+  final Duration showDuration;
+  final Widget child;
+
+  TooltipState createState() => new TooltipState();
+}
+
+class TooltipState extends State<Tooltip> {
+
+  Performance _performance;
+  OverlayEntry _entry;
+  Timer _timer;
+
+  void initState() {
+    super.initState();
+    _performance ??= new Performance(duration: config.fadeDuration)
+      ..addStatusListener((PerformanceStatus status) {
+        switch (status) {
+          case PerformanceStatus.completed:
+            assert(_entry != null);
+            assert(_timer == null);
+            resetShowTimer();
+            break;
+          case PerformanceStatus.dismissed:
+            assert(_entry != null);
+            assert(_timer == null);
+            _entry.remove();
+            _entry = null;
+            break;
+          default:
+            break;
+        }
+      });
+  }
+
+  void didUpdateConfig(Tooltip oldConfig) {
+    super.didUpdateConfig(oldConfig);
+    if (config.fadeDuration != oldConfig.fadeDuration)
+      _performance.duration = config.fadeDuration;
+    if (_entry != null &&
+        (config.message != oldConfig.message ||
+         config.backgroundColor != oldConfig.backgroundColor ||
+         config.style != oldConfig.style ||
+         config.textColor != oldConfig.textColor ||
+         config.borderRadius != oldConfig.borderRadius ||
+         config.height != oldConfig.height ||
+         config.padding != oldConfig.padding ||
+         config.opacity != oldConfig.opacity ||
+         config.verticalOffset != oldConfig.verticalOffset ||
+         config.screenEdgeMargin != oldConfig.screenEdgeMargin ||
+         config.preferBelow != oldConfig.preferBelow))
+      _entry.markNeedsBuild();
+  }
+
+  void resetShowTimer() {
+    assert(_performance.status == PerformanceStatus.completed);
+    assert(_entry != null);
+    _timer = new Timer(config.showDuration, hideTooltip);
+  }
+
+  void showTooltip() {
+    if (_entry == null) {
+      RenderBox box = context.findRenderObject();
+      Point target = box.localToGlobal(box.size.center(Point.origin));
+      _entry = new OverlayEntry(builder: (BuildContext context) {
+        TextStyle textStyle = (config.style ?? Theme.of(context).text.body1).copyWith(color: config.textColor ?? Colors.white);
+        return new _TooltipOverlay(
+          message: config.message,
+          backgroundColor: config.backgroundColor ?? Colors.grey[700],
```

Comment:
Actually upon further review I'm going to punt on this. This isn't something that, per the material spec, is configurable, and Theme doesn't actually let you configure the stuff that it exposes this way anyway (like hintColor -- it's computed on creation but you can't directly set it).